components/style/matching.rs

```diff
@@ -486,6 +484,14 @@ pub enum StyleSharingResult<ConcreteRestyleDamage: TRestyleDamage> {
     StyleWasShared(usize, ConcreteRestyleDamage, RestyleResult),
 }
 
+// Callers need to pass several boolean flags to cascade_node_pseudo_element.
+// We encapsulate them in this struct to avoid mixing them up.
+struct CascadeBooleans {
+    shareable: bool,
+    cacheable: bool,
+    animate: bool,
+}
```

Comment:
I think it'd be cleaner to use boolean enum for each one of those instead of a struct holding the three bools. Also, note that `cacheable` wasn't given as parameter to `cascade_node_pseudo_element`. Is that intentional, or just a consequence of it? It's probably ok in any case, I guess.